File: Interfaces/src/ArrayList.java
public class ArrayList implements List {

    int length = 0;
    int a [] = new int[50];


    public void showArray() {
        for(int i = 0; i < this.size(); i++){
            System.out.print(a[i] + " ");
        }
        System.out.println();
    }

    @Override
    public int size() {
        return this.length;
    }

    @Override
    public void push(int value) {
        this.a[this.size()] = value;
        length++;
    }


    @Override
    public void pop() {
        int delete = a[this.size()];
        a[this.size()] = 0;
        length--;
        //System.out.println(delete);
    }

    @Override
    public void unshift(int value) {
        for(int i = this.size(); i >  0 ; i--){
            this.a[i] = this.a[i - 1];
        }
        a[0] = value;
        length++;
    }

    @Override
    public void shift() {
        String delete = this.size() + " deleted";
        for(int i = 0; i < size() ; i++){
            this.a[i] = this.a[i + 1];
        }
        length--;
        //System.out.println(delete);
    }


    @Override
    public void toString(int[] a) {
        String str = "";
        for(int i = 0; i < size() ; i++){
            str += this.a[i] + ", ";
        }
        str =  str.substring(0, str.length() - 2) + " - String";
        System.out.println(str);
    }

    @Override
    public int[] sort(int[] arr) {
        for (int i = 0, endI = size() - 1; i < endI; i++){
            boolean isSwap = false;
            int tmp;
            for (int j = 0, endJ = endI - i; j < endJ; j++){
                if (arr[j] > arr[j + 1]){
                    tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                    isSwap = true;
                }
            }
            if (!isSwap){break;}
        }
        return arr;
    }

}

File: Js Interfaces/ES6/js/ArrayList.js
class ArrayList extends  List {
	array = [];
	size = 0;
	str = "";

	clear =  () => {
		this.array = [];
		this.size = 0;
	};
	length =  () => {
		return this.size;
	};
	push = (value) => {
		this.array[this.size] = value;
		this.size++;
		return value;
	};
	pop = () => {
		let lastElem = this.array[this.size - 1];
		let a = [];
		for(let i = 0; i < this.size - 1; i++){
			a[i] = this.array[i];
		}
		this.array = a;
		this.size--;
		return lastElem;
	};
	shift = () => {
		let firstElem = this.array[0];
		let a = [];
		for(let i = 0; i < this.size - 1; i++){
			a[i] = this.array[i + 1];
		}
		this.array = a;
		this.size--;
		return firstElem;
	};
	unshift = (value) =>{
		let a = [];
		a[0] = value;
		for(let i = 0; i < this.size; i++){
			a[i + 1] = this.array[i];
		}
		this.array = a;
		this.size++;
		return value;
	};
	toString = () => {
		for(let i = 0; i < this.size; i++){
			this.str +=this.array[i] + ", ";
		}
		this.str = this.str.substring(0, this.str.length - 2);
		return this.str;
	}
	sort = (f) => {
		if(f && (typeof f == "function" )){
			for(let i = 0; i < this.size; i++){
				for(let j = 0; j < this.size; j++){
					if(j === this.size - 1){
						break;
					}
					if(f(this.array[j], this.array[j+1]) === 	1){
						[this.array[j], this.array[j+1]] = [this.array[j+1], this.array[j]];
					}
				}
			}
		}else{
			for(let i = 0; i < this.size; i++){
				for(let j = 0; j < this.size; j++){
					if(j === this.size - 1){
						break;
					}if( String(this.array[j]) > String(this.array[j + 1]) ){
						[this.array[j], this.array[j+1]] = [this.array[j+1], this.array[j]];
					}
				}
			}
		}
	}
}